### PR TITLE
re-write exp and log functions

### PR DESCRIPTION
**Pull request by vtavana**
The PR changes the implementation of `dpnp.exp`, `dpnp.expm1`, `dpnp.log10`, `dpnp.log1p`, and `dpnp.log2`. These functions now invoke `oneapi::mkl::vm` implementation from pybind11 extension of OneMKL VM if possible or uses `dpctl.tensor` implementation if not.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/backend/extensions/vm/types_matrix.hpp [hunk 1]
```diff
@@ -276,6 +276,36 @@ struct DivOutputType
...
+ * @brief A factory to define pairs of supported types for which
+ * MKL VM library provides support in oneapi::mkl::vm::exp<T> function.
+ *
+ * @tparam T Type of input vector `a` and of result vector `y`.
+ */
+template <typename T>
+struct ExpOutputType
+{
+    using value_type = typename std::disjunction<
+        dpctl_td_ns::TypeMapResultEntry<T, double, double>,
```
Comment (antonwolfy) [inline dpnp/backend/extensions/vm/types_matrix.hpp:289]: what is about complex types? they are also supported by MKL

Comment (antonwolfy) [inline dpnp/backend/extensions/vm/types_matrix.hpp:289]: result type can be omitted when the same as source type
```suggestion
        dpctl_td_ns::TypeMapResultEntry<T, double>,
```

File: tests/test_umath.py [hunk 2]
```diff
@@ -509,39 +509,46 @@ def test_invalid_shape(self, shape):
 
 
 class TestExp:
-    def test_exp(self):
-        array_data = numpy.arange(10)
-        out = numpy.empty(10, dtype=numpy.float64)
+    @pytest.mark.parametrize(
+        "dtype", get_all_dtypes(no_bool=True, no_complex=True)
+    )
+    def test_exp(self, dtype):
```
Comment (antonwolfy) [inline tests/test_umath.py:515]: Could you please add `test_exp_complex`?

File: dpnp/backend/extensions/vm/types_matrix.hpp [hunk 3]
```diff
@@ -325,6 +355,55 @@ struct LnOutputType
...
+};
+
+/**
+ * @brief A factory to define pairs of supported types for which
+ * MKL VM library provides support in oneapi::mkl::vm::log10<T> function.
+ *
+ * @tparam T Type of input vector `a` and of result vector `y`.
+ */
+template <typename T>
+struct Log10OutputType
```
Comment (antonwolfy) [inline dpnp/backend/extensions/vm/types_matrix.hpp:395]: should be placed before `Log1pOutputType` for lexicographical order

File: dpnp/backend/extensions/vm/vm_py.cpp [hunk 4]
```diff
@@ -79,9 +84,14 @@ static unary_impl_fn_ptr_t conj_dispatch_vector[dpctl_td_ns::num_types];
...
 static unary_impl_fn_ptr_t cosh_dispatch_vector[dpctl_td_ns::num_types];
 static binary_impl_fn_ptr_t div_dispatch_vector[dpctl_td_ns::num_types];
+static unary_impl_fn_ptr_t exp_dispatch_vector[dpctl_td_ns::num_types];
+static unary_impl_fn_ptr_t expm1_dispatch_vector[dpctl_td_ns::num_types];
 static unary_impl_fn_ptr_t floor_dispatch_vector[dpctl_td_ns::num_types];
 static binary_impl_fn_ptr_t hypot_dispatch_vector[dpctl_td_ns::num_types];
 static unary_impl_fn_ptr_t ln_dispatch_vector[dpctl_td_ns::num_types];
+static unary_impl_fn_ptr_t log1p_dispatch_vector[dpctl_td_ns::num_types];
+static unary_impl_fn_ptr_t log2_dispatch_vector[dpctl_td_ns::num_types];
+static unary_impl_fn_ptr_t log10_dispatch_vector[dpctl_td_ns::num_types];
```
Comment (antonwolfy) [inline dpnp/backend/extensions/vm/vm_py.cpp:94]: the same, should be before `log1p_dispatch_vector`

File: dpnp/dpnp_iface_trigonometric.py [hunk 5]
```diff
@@ -57,8 +57,13 @@
     dpnp_atanh,
     dpnp_cos,
     dpnp_cosh,
+    dpnp_exp,
+    dpnp_expm1,
     dpnp_hypot,
     dpnp_log,
+    dpnp_log1p,
+    dpnp_log2,
+    dpnp_log10,
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_trigonometric.py:66]:
```suggestion
    dpnp_log10,
    dpnp_log2,
```

Comment (vtavana) [reply dpnp/dpnp_iface_trigonometric.py:66]: `pre-commit` (`isort (python)`) automatically changes the order to the current one here.

Comment (antonwolfy) [reply dpnp/dpnp_iface_trigonometric.py:66]: I see, then please ignore my comment around here

**Review by antonwolfy — APPROVED** [review]
Thank you @vtavana

**Pull request merged**